src/ast.scm

```diff
@@ -310,26 +310,30 @@
        (symbol? (cadr e))))
 
 (define (lam:args x) (cadr x))
-(define (lam:argnames x) (llist-vars (lam:args x)))
+(define (lam:argnames x) (llist-vars (lam:args x) #f))
 (define (lam:vinfo x) (caddr x))
 (define (lam:body x) (cadddr x))
 (define (lam:sp x) (cadddr (lam:vinfo x)))
 
-(define (bad-formal-argument v)
-  (error (string #\" (deparse v) #\" " is not a valid function argument name")))
+(define (bad-formal-argument v fname)
+  (error (string #\" (deparse v) #\" " is not a valid function argument name"
+                 (if fname
+                     (string " inside the function definition for \"" fname "\"")
+                     "")
+                 ". Perhaps you meant to write \"==\" instead of \"=\"")))
```

Comment:
We should try not to show this part if `function` was used instead of `=` syntax.

Reply:
Yeah, I've been thinking about moving this check to an earlier stage to also give useful names for function definitions like `(::Foo)(bar(a)) = 42` as well. Currently we can't show the `::Foo` part because it's not preserved until this stage in lowering.